text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+at the location specified by `BP_DEPENDENCY_METADATA`, which defaults to
+`/platform/deps/metadata` (the intent is to pilot and try out this or possibly
+other locations, eventually proposing an RFC with Cloud-Native buildpacks to
+standardize the location).
+
+Buildpacks do not care if there are multiple sources of metadata information,
+however, this needs to be merged and presented to the buildpack as one single
+directory. How that information is merged is outside of the scope of this
+document, but the directory structure defined in [Metadata
+Format](#metadata-format) guarantees that there will not be any duplicate
+dependency ids.
+
+This document does not specify how the dependency metadata folder should be
+provided to buildpacks but here are some possibilities:
+
+1. External. The metadata information can be managed outside of the buildpack
+   lifecycle. This allows for users to manually pull in the metadata they would
+   like when they would like it. The metadata can then be mapped into the build
+   container at the specified location using a volume mount to `pack build`.
+1. Via buildpack. We could add a Paketo buildpack to the beginning of buildpack
+   order groups that is responsible for pulling metadata and overrides
+   `BP_DEPENDENCY_METADATA` to point to its layer. The buildpack can check for
+   dependency metadata updates when it runs.
+1. Via builder. The builder could come with metadata included under
+   `/platform/deps/metadata`. This would likely require a builder with many
+   buildpacks to update very frequently though, so may not be the best idea for
+   the average case, however, it could be very useful as a way to distribute
+   dependencies in offline environments.
+1. Via the platform. Platforms may choose to offer enhanced functionality to
+   more easily distribute dependencies. In the end, the platform just needs to
+   ensure that the dependency metadata is available to the buildpack in the
+   required location.
+
+In addition to the flexibility of where the dependencies originate, this
+proposal also provides flexibility in how those dependencies are managed such
+as floating them so the latest versions are always available, pinning to a
+specific set of dependencies or even pinning and including the dependencies
+with the metadata.
+
+## Dependency Assets
+This proposal does not impact how dependencies are being distributed and it
+supports the current methods referring to remote locations, like upstream
+projects, or using the Deps Server. The dependency metadata just needs to point
+to the location from which the dependency should be fetched as is currently
+being done with the buildpack.toml dependency metadata.
+
+For builders, buildpacks, or platforms that would like to inject dependency
+assets directly into the build container, perhaps to support offline builds, we
+propose defining `BP_DEPENDENCY_BINARIES` which defaults to
```

Comment:
I think this environment variable name should more clearly indicate that it is referring to a filepath prefix. Maybe something `BP_DEPENDENCY_BIN_PATH`?

Reply:
Maybe `BP_DEPENDENCY_ASSET_ROOT` or `_PATH`?

Reply:
Yeah, both of those work as well. Do we have examples of prior art for naming env vars for directories - both in Paketo and in the upstream Buildpacks projects?

Reply:
I am trying out `BP_DEPENDENCY_ASSET_ROOT` for now.